### PR TITLE
Transition types as members

### PR DESCRIPTION
**Pull request by hroest**
- Stores transition types (detecting, identifying and quantifying) as members in ReactionMonitoringTransition instead of meta values  (saves space)
- does not output user values to TraML if the default is true for all values
- provide easy interface to query type of transition

### CODE REVIEW

**Comment by hroest** [comment]
@grosenberger  can you have a look at this set of changes?

**Comment by hroest** [comment]
on a typical TraML file such as `src/tests/topp/ConvertTSVToTraML_4_output.TraML` the optimization of not writing out the params each time also saves ca 25% of disk space (or 10% in compressed format).

File: src/tests/topp/OpenSwathAssayGenerator_output.TraML [hunk 1]
```diff
@@ -61,9 +61,6 @@
       <cvParam cvRef="MS" accession="MS:1001226" name="product ion intensity" value="10000"/>
       <cvParam cvRef="MS" accession="MS:1002007" name="target SRM transition"/>
       <userParam name="annotation" type="xsd:string" value="y7^2"/>
-      <userParam name="detecting_transition" type="xsd:string" value="true"/>
-      <userParam name="identifying_transition" type="xsd:integer" value="0"/>
-      <userParam name="quantifying_transition" type="xsd:integer" value="1"/>
```
Comment (hroest) [inline src/tests/topp/OpenSwathAssayGenerator_output.TraML]: was this valid input? I now changed it everywhere to "xsd:boolean" and true/false

File: src/openms/source/FORMAT/HANDLERS/TraMLHandler.cpp [hunk 2]
```diff
@@ -1564,7 +1581,25 @@ namespace OpenMS
       }
       else if (parent_tag == "Transition")
       {
-        actual_transition_.setMetaValue(name, data_value);
+        // see xsd:boolean reference (http://books.xmlschemata.org/relaxng/ch19-77025.html)
+        // The value space of xsd:boolean is true and false. Its lexical space
+        // accepts true, false, and also 1 (for true) and 0 (for false).
+        if (name == "detecting_transition")
+        {
+          actual_transition_.setDetectingTransition((value == "true" || value == "1"));
```
Comment (hroest) [inline src/openms/source/FORMAT/HANDLERS/TraMLHandler.cpp]: handles "1" and "true" as `true`

File: src/tests/topp/OpenSwathAssayGenerator_output.TraML (hunk 1 above)
Comment (grosenberger) [inline src/tests/topp/OpenSwathAssayGenerator_output.TraML]: I think this was outdated. Thanks for changing!

**Comment by grosenberger** [comment]
Wow, this was super fast and looks great! Thank you very much!

**Comment by hroest** [comment]
does it work well? can you tell whether it improves TraML handling  / memory performance?

File: src/tests/topp/OpenSwathAssayGenerator_output.TraML (hunk 1 above)
Comment (hroest) [inline src/tests/topp/OpenSwathAssayGenerator_output.TraML]: well it now accepts input as true/false and as 1/0 but I like true/false better

**Pull request merged**